app/models/behaviour.rb

```diff
@@ -235,13 +270,63 @@ def translation_keys
       else
         raise NotImplementedError
       end
+    when "decision_tree"
+      [
+        translation_key("explanation",    "Skill explanation"),
+        translation_key("clarification",  "Clarification message"),
+        translation_key("keywords",       "Valid keywords (comma separated)"),
+        build_array_from_tree(config["tree"]["nodes"], config["tree"]["initial"], 1)
+      ].flatten
     else
       raise NotImplementedError
     end
   end
 
   private
 
+  def build_array_from_tree(nodes, initial_uuid, level)
+    translations = []
+    uuids = [[initial_uuid, level]]
+    until uuids.empty?
+      uuid, level = uuids.shift
+      translations.push(translation_key("tree/nodes/#{uuid}/message", "#{level}#{level.ordinal} Question"))
+      translations.concat(
+        nodes[uuid]['options'].map.with_index do |option, ix|
+          uuids.push([option['next'], level + 1]) unless option['next'].nil?
+          translation_key("tree/nodes/#{uuid}/options/[next=#{option['next']}]/label", "Option")
+        end
+      )
+    end
+
+    return translations
+  end
+
+  def build_manifest_tree(nodes, uuid)
+
+    result = {
+      id: uuid
+    }
+    loc_value = localized_value("tree/nodes/#{uuid}/message")
+
+    options = nodes[uuid]['options']
+    if options.empty?
+      result[:answer] = loc_value
+    else
+      result[:question] = loc_value
+      result[:responses] =  options.map do |option|
+                              next_uuid = option['next']
+                              {
+                                keywords: localized_value("tree/nodes/#{uuid}/options/[next=#{next_uuid}]/label") do |keywords|
+                                  keywords.split(/,\s*/)
+                                end,
+                                next: build_manifest_tree(nodes, next_uuid)
```

Comment:
Add a technical debt to validate that the tree in the manifest is a tree a not a graph so there is no danger of infinite recursion here and in other places.